bigquery/google/cloud/bigquery/job.py

```diff
@@ -2475,237 +2475,268 @@ def __eq__(self, other):
+
+        'write_ms_avg': 'writeMsAvg',
+        'write_ms_max': 'writeMsMax',
+        'write_ratio_avg': 'writeRatioAvg',
+        'write_ratio_max': 'writeRatioMax',
+        'records_read': 'recordsRead',
+        'records_written': 'recordsWritten',
+        'status': 'status',
+        'shuffle_output_bytes': 'shuffleOutputBytes',
+        'shuffle_output_bytes_spilled': 'shuffleOutputBytesSpilled',
+        'steps': 'steps',
+     }
 
-    :type parallel_inputs: int
-    :param completed_parallel_inputs: number of parallel input
-            segments in the stage
+    def __init__(self):
+        self._properties = {}
 
-    :type completed_parallel_inputs: int
-    :param completed_parallel_inputs: number of parallel input
-            segments completed
+    @classmethod
+    def from_api_repr(cls, resource):
+        """Factory: construct instance from the JSON repr.
 
-    :type wait_ms_avg: int
-    :param wait_ms_avg: milliseconds the average shard spent
-            waiting to be scheduled
+        Args:
+            resource(Dict[str: object]):
+                ExplainQueryStage representation returned from API
 
-    :type wait_ms_max: int
-    :param wait_ms_max: milliseconds the slowest shard spent
-            waiting to be scheduled
+        Returns:
+            google.cloud.bigquery.QueryPlanEntry:
+                Query plan entry parsed from ``resource``
+        """
+        entry = cls()
+        entry._properties = copy.deepcopy(resource)
+        return entry
 
-    :type wait_ratio_avg: float
-    :param wait_ratio_avg: average wait ratio
+    @property
+    def name(self):
+        """Union[str, None]: Human-readable name of the stage."""
+        return self._properties.get('name')
 
-    :type wait_ratio_max: float
-    :param wait_ratio_max: maximum wait ratio
+    @property
+    def entry_id(self):
+        """Union[str, None]: Unique ID for the stage within the plan."""
+        return self._properties.get('id')
 
-    :type read_ms_avg: int
-    :param read_ms_avg: milliseconds the average shard spent
-            reading input
+    @property
+    def start(self):
+        """Union[Datetime, None]: Datetime when the stage started."""
+        if self._properties.get('startMs') is not None:
+            return _datetime_from_microseconds(
+                    self._properties.get('startMs') * 1000.0)
 
-    :type read_ms_max: int
-    :param read_ms_max: milliseconds the slowest shard spent
-            reading input
+    @property
+    def end(self):
+        """Union[Datetime, None]: Datetime when the stage ended."""
+        if self._properties.get('endMs') is not None:
+            return _datetime_from_microseconds(
+                    self._properties.get('endMs') * 1000.0)
 
-    :type read_ratio_avg: float
-    :param read_ratio_avg: average read ratio
+    @property
+    def input_stages(self):
+        """List(int): Entry IDs for stages that were inputs
+                for this stage.
```

Comment:
We don't do a hanging indent for property (I think Sphinx won't render it right). Should be

```
"""List(int): Entry IDs for stages that were inputs
for this stage.
```

 See: https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html#example-google